modules/apps/transfer/keeper/relay.go

```diff
@@ -109,6 +109,16 @@ func (k Keeper) sendTransfer(
 		); err != nil {
 			return 0, err
 		}
+
+		denomTrace := types.ParseDenomTrace(fullDenomPath)
+		if denomTrace.GetPath() == "" {
```

Comment:
I believe this check is needed because you could have a denom like `transfer/channel-0/uatom` and send it on `channel-1` and then `SenderChainIsSource` will still pass, but the denom is not native. Does it make sense?

Reply:
@crodriguezvega is correct. `SenderChainIsSource` is just determining if the tokens being sent originate from the sending chain or if they originate from the receiving chain (path unwinding)

The terminology `native token` should imply the token is not an IBC token or another token that was originally transferred from another chain. An IBC token is not native, but sender is source when it is being sent on a channel it didn't arrive from. Tokens are native regardless of where they are being transferred to

Reply:
aaah this makes sense. Just to clarify in the example below 
```
chain A (channel 0) -> (channel-0) chain B (channel-1) -> (channel-1) chain A
stake                  transfer/channel-0/stake           transfer/channel-1/transfer/channel-0/stake
```

if we donot have this `GetPath()` check ChainB ---> ChainA  transfer will pass the function `SenderChainIsSource` but it is not a native denom right?

Reply:
Yes, that's right. When sending tokens of denom `transfer/channel-0/stake` from chain B, you have these two situations:

- If you send them over `channel-0` (the channel from where they were received, since `channel-0` is part of the trace path of the denom), then `SenderChainIsSource` will return false, since it is the receiving chain (chain A) the source of the tokens.
- If you send them over a different channel (for example `channel-1`), then chain B becomes the source chain and `SenderChainIsSource` will return true. In this case, and assuming a definition of native tokens that does not include IBC denoms, then we should not store in state the total escrow for these tokens.

I hope this helps to clarify!

Reply:
Thank you! this makes sense